public_html/wp-content/plugins/wordcamp-reports/views/public/payment-activity.php

```diff
@@ -47,7 +47,14 @@
 
 		<div class="field_wordcamp-id">
 			<label for="wordcamp-id">WordCamp <span>(optional)</span></label>
-			<?php echo get_wordcamp_dropdown( 'wordcamp-id', array(), $wordcamp_id ); ?>
+			<?php if ( ! isset( $_GET[ 'action' ] ) ) {
+				// Show only latest 100 WordCamps by default
+				echo get_wordcamp_dropdown( 'wordcamp-id', array( 'posts_per_page' => 100, 'order' => 'DESC' ), $wordcamp_id );
+			} else {
+				// If something is being searched, return WordCamps for that year, or the default latest year, to reduce memory usage
+				$wc_year = ( isset( $years ) ) ? $years : date( 'Y' );
+				echo get_wordcamp_dropdown( 'wordcamp-id', array( 'date_query' => array( 'year' => $year, 'compare' => '=' ) ), $wordcamp_id ); 
+			} ?>
```

Comment:
So if someone wanted to view WCUS 2018 info, would they have to do this?

1. Select `2018`
2. Select `Entire Year`
3. Press `Show Results`
4. Select the camp from the dropdown now that it's there
5. Press `Show Results` again

I'm guessing most folks won't realize they need to do that. We might need to add some instructions, or consider a more intuitive flow.

If [caching](https://github.com/WordPress/wordcamp.org/issues/816#issuecomment-1397354654) would solve the problem without an UI/UX changes, I think that'd be preferable.

Reply:
> If https://github.com/WordPress/wordcamp.org/issues/816#issuecomment-1397354654 would solve the problem

let me re-look at a caching option and see what that looks like.

I'm unsure what the normal workflow here is (presumably it's in stats somewhere), although it does suggest that the dropdown is optional, and also if you search by a single year, you do get different year results - presumably based on when the site was created?

Reply:
👍🏻 

If caching doesn't help, another option would be to refactor the flow. instead of populating the dropdown during the initial page load, we could do it async via XHR every time the year or time period changes. Then the XHR could only fetch the camps during that year.

Actually, just fetching the data via XHR without any year filtering might be enough for now, since the data needed to compile the page would be split across two processes. We may eventually reach a point where we can't show all camps, but I'd be surprised if we're anywhere close to that with only `1400` sites. 

🤔 That makes me think that [reducing the data](https://github.com/WordPress/wordcamp.org/issues/816#issuecomment-1398465282) is worth another look, since it seems like the best option if it works.